SmartHunter/Core/Config/ConfigContainer.cs

```diff
@@ -88,17 +91,27 @@ public void Save()
 
             try
             {
-                var settings = new JsonSerializerSettings();
-                settings.Formatting = Formatting.Indented;
-                settings.NullValueHandling = NullValueHandling.Ignore;
-                settings.ContractResolver = new ContractResolver();
+                var settings = new JsonSerializerSettings
+                {
+                    Formatting = Formatting.Indented,
+                    NullValueHandling = NullValueHandling.Ignore,
+                    ContractResolver = new ContractResolver()
+                };
 
                 settings.Converters.Add(new StringEnumConverter());
                 settings.Converters.Add(new StringFloatConverter());
 
                 var jsonString = JsonConvert.SerializeObject(Values, settings);
 
-                File.WriteAllText(FullPathFileName, jsonString);
+                using (var fileStream = new FileStream(FileName, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Write, 4096, FileOptions.None))
```

Comment:
The issue here is that it fails to resolve the file stream correctly. So we need to explicitly grab it rather than relying on File.WriteAllText to auto resolve it for us